src/Plugin/Field/FieldWidget/ProductVariationSkuWidget.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Drupal\commerce\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\StringTextfieldWidget;
+
+/**
+ * Plugin implementation of the 'commerce_auto_sku' widget.
+ *
+ * @FieldWidget(
+ *   id = "commerce_auto_sku",
+ *   label = @Translation("Commerce auto SKU"),
+ *   field_types = {
+ *     "string"
+ *   }
+ * )
+ */
+class ProductVariationSkuWidget extends StringTextfieldWidget {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return array(
+      'iniqid_enabled' => TRUE,
+      'more_entropy' => FALSE,
+      'prefix' => 'sku-',
+      'suffix' => '',
+      'size' => 60,
+      'placeholder' => '',
+    ) + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $none = t('None');
```

Comment:
As I understand $this->t('None') is a shortcut to this:

`t('None', [], ['context' => get_class($this)]);`

What does it mean "None" will never accept global translations but only translations made in the given context. That is not the case for english word "None" but not sure here because I translate it to my own language and for me it looks like as a ***global*** notation.